sql/mysqld.cc

```diff
@@ -8743,7 +8743,7 @@ static int fix_paths(void)
   my_realpath(mysql_home,mysql_home,MYF(0));
   /* Ensure that mysql_home ends in FN_LIBCHAR */
   pos=strend(mysql_home);
-  if (pos[-1] != FN_LIBCHAR)
+  if (pos==mysql_home || pos[-1] != FN_LIBCHAR)
```

Comment:
What is being fixed here? How is it related to encryption?

Reply:
If mysql_home is empty, pos[-1] is a buffer underflow - which happens in one of the encryption testcases. I'll mention this in the commit message.